examples/phi3/phi3_vision.py

```diff
@@ -0,0 +1,362 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+import argparse
+import glob
+import json
+import os
+import readline
+from pathlib import Path
+
+from olive.common.utils import run_subprocess
+from olive.workflows import run as olive_run
+from olive.workflows.run.config import RunConfig
+
+# flake8: noqa: T201
+# phi3-vision only supports CPU and CUDA targets for now
+TARGETS = ["cpu", "cuda"]
+config_path = Path(__file__).parent / "vision" / "config_templates"
+
+
+def get_args(raw_args):
+    parser = argparse.ArgumentParser(description="phi3 optimization")
+
+    parser.add_argument(
+        "--target",
+        type=str,
+        default="cpu",
+        required=False,
+        choices=TARGETS,
+        help=f"Choose from {TARGETS}",
+    )
+    parser.add_argument(
+        "--precision",
+        type=str,
+        default="int4",
+        choices=["int4"],
+        help=("Precision of quantized model. Currently only int4 is supported. "),
+    )
+    parser.add_argument(
+        "--inference",
+        action="store_true",
+        help="Run inference with optimized model",
+    )
+    parser.add_argument(
+        "--optimized_model_path",
+        type=str,
+        help="Run inference with optimized model",
+    )
+    parser.add_argument(
+        "--output_dir",
+        type=str,
+        default="cache/phi3-vision-128k-instruct",
+        required=False,
+        help="Path to folder to store ONNX model and additional files (e.g. GenAI config, external data files, etc.)",
+    )
+    parser.add_argument(
+        "--cache_dir",
+        required=False,
+        default="cache",
+        help="Path to cache directory",
+    )
+
+    return parser.parse_args(raw_args)
+
+
+def is_model_ready(input_model_path):
+    if not input_model_path.exists():
+        return False
+    try:
+        from transformers import AutoConfig, AutoModelForCausalLM, AutoProcessor
+
+        config = AutoConfig.from_pretrained(input_model_path, trust_remote_code=True)
+        processor = AutoProcessor.from_pretrained(input_model_path, trust_remote_code=True)
+        model = AutoModelForCausalLM.from_pretrained(input_model_path, trust_remote_code=True)
+        del model, processor, config
+    except Exception as e:
+        print(f"Error loading model: {e}")
+        return False
+    return True
+
+
+def build_text_embedding(args, input_model_path):
```

Comment:
Is this manual building of the model necessary? I am able to generate the same graph using onnx conversion of the embed_tokens submodule https://github.com/microsoft/Olive/commit/777d82250e8a858b1fe584aa90e6b5b56164b478

Reply:
https://huggingface.co/microsoft/Phi-3-vision-128k-instruct-onnx-cpu/blob/main/onnx/builder.py
It is borrowed from huggingface builder.py which is mentined in genai's tutorial as well.
I am not sure if it is totally same with build_text_embedding.

Reply:
Since the graph looks the same, could you try the model generated from the config. 

It feels unnecessary and non standard to manually generate the graph is a simple exported model works.

Reply:
<img width="782" alt="image" src="https://github.com/user-attachments/assets/83c45b56-59d0-4bcf-824b-f0c1cab759fc">
Conversion cannot not work directly. Seems the generator requires the inputs name of `inputs_embeds`.

Reply:
It worked. Just updated.